### PR TITLE
#4339 collection popover

### LINKED ISSUE #4339: Hover card collection view
- **collection view**
  - hint collection numbers https://beta.kodadot.xyz/rmrk/collection/0a8ce195286c168f19-DONKEY
  - you want to know more about particular collection statistics
  - unique owners
  - highest sale
  - floor price
  - total volume in base currency
  - 3 pictures with highest sale/offer

Designs can be found in #4117

The card should show up, for example, when user hovers at NFT item over the collection link

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/4117#issuecomment-1285588068_

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4339
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2023-01-15 at 5 25 34 PM

### CODE REVIEW

File: components/collection/utils/useCollectionDetails.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        listedCount: data.value.stats.listed.length,
+        collectionLength: data.value.stats.base.length,
+        collectionFloorPrice: Math.min(
+          ...data.value.stats.listed.map((item) => parseInt(item.price))
+        ),
+        uniqueOwnersPercent: `${
+          (uniqueOwnerCount / differentOwnerCount) * 100
+        }%`,
+        collectionDailyTradedVolumeNumber: getVolume(
+          data.value.stats.sales.map((nft) => nft.events).flat()
```
Comment (Jarsen136) [inline components/collection/utils/useCollectionDetails.ts:48]: The result is not equal to the `DailyTradedVolume`? It's the total traded volume as I see in the graphql query.

`components/rmrk/Gallery/CollectionActivity.vue` the method at  line183  would help.

**Review by Jarsen136 — CHANGES_REQUESTED** [review]

**Comment by Jarsen136** [comment]
It would be better to have white colour of border for images in darkmode. 
image

**Comment by yangwao** [comment]
> It would be better to have white colour of border for images in darkmode.

@exezbcz 👀

**Comment by exezbcz** [comment]
> > It would be better to have white colour of border for images in darkmode.
> 
> @exezbcz 👀

Yes, it should be like that 🚀

File: components/collection/utils/useCollectionDetails.ts (hunk 1 above)
Comment (prachi00) [reply components/collection/utils/useCollectionDetails.ts:48]: updated

File: components/collection/utils/useCollectionDetails.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      limit: 3,
+      orderBy: 'price_DESC',
+      collectionId: collectionId,
+      where: {
+        collection: { id_eq: collectionId },
+      },
+    },
+  })
+
+  watch(data as unknown as NFTListSold, (list) => {
```
[the remark this answers is not in the document]
Comment (prachi00) [reply components/collection/utils/useCollectionDetails.ts:93]: I dont think we can refactor it

[the remark this answers is not in the document]
Comment (Jarsen136) [reply components/collection/utils/useCollectionDetails.ts:93]: We could try to reuse `useIdentitySoldData` in `components/identity/utils/useIdentity.ts` to refactor it

[the remark this answers is not in the document]
Comment (prachi00) [reply components/collection/utils/useCollectionDetails.ts:93]: we could, query is a bit different though, wouldn't it be more cleaner to not reuse it in this case?

[the remark this answers is not in the document]
Comment (prachi00) [reply components/collection/utils/useCollectionDetails.ts]: okay, I reused, gave an optional param for collectionid

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by exezbcz** [comment]
we should also add it on the nft card hover 
- when you hover collection name on the card

image


also the highest sales dont look they are correct
image
context: https://deploy-preview-4714--koda-nuxt.netlify.app/rmrk/gallery/16219762-7E982817A2D2575031-KDMN-KODAMONEY_008-0000000000000009?page=1
collection: 
https://deploy-preview-4714--koda-nuxt.netlify.app/rmrk/collection/7E982817A2D2575031-KDMN?page=1

**Comment by prachi00** [comment]
@exezbcz wouldnt that open 2 popovers, one for nft card and other for collection, is that okay?

**Comment by prachi00** [comment]
> 

oh, let me check

**Comment by prachi00** [comment]
@exezbcz fixed the query and added popover to cards

**Comment by prachi00** [comment]
> Code Climate has analyzed commit [530f571](https://github.com/kodadot/nft-gallery/commit/530f571b1cafc5567e8780ab13c9264fedb76f39) and detected **2 issues** on this pull request.
> 
> Here's the issue category breakdown:
> 
> Category	Count
> Duplication	2
> View more on [Code Climate](https://codeclimate.com/github/kodadot/nft-gallery/pull/4714).

^ I made a new query because the one we use in identity is a bit different, with different params and we also dont need all the stuff it is giving back

**Comment by prachi00** [comment]
@exezbcz can you confirm if it looks fine now?

**Comment by exezbcz** [comment]
please make the time visible after hovering out lower, like on the user hover card
image

activation on the nft cards is bigger than should 
image
- it activates in the middle of the vertical row. - should only popup on the text

**Comment by prachi00** [comment]
> please make the time visible after hovering out lower, like on the user hover card image
> 
> activation on the nft cards is bigger than should image
> 
> * it activates in the middle of the vertical row. - should only popup on the text

I dont understand, time visible?

**Comment by prachi00** [comment]
Screen Shot 2023-01-20 at 10 32 10 PM
@preschian @Jarsen136 @roiLeo how can we have unique owner 1 and total owner 0? isnt this wrong, I calculated the % using these and it says infinity
https://beta.kodadot.xyz/bsx/collection/2551182625

**Comment by prachi00** [comment]
I modified the logic a bit
`       (uniqueOwnerCount / (uniqueOwnerCount + differentOwnerCount)) * 100
`       
I am hoping this would give the correct value always

**Comment by prachi00** [comment]
@exezbcz please confirm if it looks okay now :)

**Comment by exezbcz** [comment]
> @exezbcz please confirm if it looks okay now :)

yup!

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 35.88 USD/KSM ~ 1.394 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0x3ca153ae1bbc364a8ce3bb25cda40093e1a4f7ee4563a6229ffdfd355dbea57a](https://kusama.subscan.io/extrinsic/0x3ca153ae1bbc364a8ce3bb25cda40093e1a4f7ee4563a6229ffdfd355dbea57a)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**